src/status_im2/contexts/chat/lightbox/zoomable_image/utils.cljs

```diff
@@ -4,9 +4,92 @@
+    (anim/animate pan-x c/init-offset duration)
+    (anim/set-val pan-x-start c/init-offset)))
 
+(defn center-y
+  [{:keys [pinch-y pinch-y-start pan-y pan-y-start]} exit?]
+  (let [duration (if exit? 100 c/default-duration)]
+    (anim/animate pinch-y c/init-offset duration)
+    (anim/set-val pinch-y-start c/init-offset)
+    (anim/animate pan-y c/init-offset duration)
+    (anim/set-val pan-y-start c/init-offset)))
+
+(defn reset-values
+  [exit? animations {:keys [focal-x focal-y]}]
+  (center-x animations exit?)
+  (center-y animations exit?)
+  (reset! focal-x nil)
+  (reset! focal-y nil))
+
+(defn rescale-image
+  [value
+   exit?
+   {:keys [x-threshold-scale y-threshold-scale]}
+   {:keys [scale saved-scale] :as animations}
+   {:keys [pan-x-enabled? pan-y-enabled?] :as props}]
+  (anim/animate scale value (if exit? 100 c/default-duration))
+  (anim/set-val saved-scale value)
+  (when (= value c/min-scale)
+    (reset-values exit? animations props))
+  (reset! pan-x-enabled? (> value x-threshold-scale))
+  (reset! pan-y-enabled? (> value y-threshold-scale)))
+
+;;; Handlers
+(defn handle-orientation-change
+  [curr-orientation
+   focused?
+   {:keys [landscape-scale-val x-threshold-scale y-threshold-scale]}
+   {:keys [rotate rotate-scale scale] :as animations}
+   {:keys [pan-x-enabled? pan-y-enabled?]}]
+  (let [duration (when focused? c/default-duration)]
+    (cond
+      (= curr-orientation orientation/landscape-left)
+      (do
+        (anim/animate rotate "90deg" duration)
+        (anim/animate rotate-scale landscape-scale-val duration))
+      (= curr-orientation orientation/landscape-right)
+      (do
+        (anim/animate rotate "-90deg" duration)
+        (anim/animate rotate-scale landscape-scale-val duration))
+      (= curr-orientation orientation/portrait)
+      (do
+        (anim/animate rotate c/init-rotation duration)
+        (anim/animate rotate-scale c/min-scale duration)))
+    (center-x animations false)
+    (center-y animations false)
+    (reset! pan-x-enabled? (> (anim/get-val scale) x-threshold-scale))
+    (reset! pan-y-enabled? (> (anim/get-val scale) y-threshold-scale))))
+
+(defn handle-exit-lightbox-signal
+  "On ios, when attempting to navigate back while zoomed in, the shared-element transition animation
+   doesn't execute properly, so we need to zoom out first"
+  [exit-lightbox-signal index scale rescale set-full-height?]
+  (when (= exit-lightbox-signal index)
+    (reset! set-full-height? false)
+    (if (> scale c/min-scale)
+      (do
+        (rescale c/min-scale true)
+        (js/setTimeout #(rf/dispatch [:navigate-back]) 70))
+      (rf/dispatch [:navigate-back]))
+    (js/setTimeout #(rf/dispatch [:chat.ui/exit-lightbox-signal nil]) 500)))
+
+(defn handle-zoom-out-signal
+  "Zooms out when pressing on another photo from the small bottom list"
+  [zoom-out-signal index scale rescale]
+  (when (and (= zoom-out-signal index) (> scale c/min-scale))
+    (rescale c/min-scale true)))
```

Comment:
I use `constants` for the main constants file `status-im2.constants`. But what if this main constants file, and a local constants are being imported into the same file, how should we name it? (although it is currently not the case)

Reply:
> Please ignore the animations file, it should be temporary. (also, it is not a new file, I just renamed it)

Okay :)

> I use constants for the main constants file status-im2.constants. But what if this main constants file, and a local constants are being imported into the same file, how should we name it? (although it is currently not the case)

Correct, and what if there's another namespace *constants* and you already used `c` in the same file? How would you name the second *constants* alias, `c2`? It's the same problem in disguise. I think the idea is to have a convention to cover well 80% of the cases (Pareto principle), but every convention breaks at some point, as you just shared an example.

Reply:
@ilmotta changed it to use `constants` for now

Reply:
These cases that we find in PR reviews, how can we make them visible to the rest of the team to reach an agreement, and not get lost (and have a discussion away from PRs)? Open a discussion maybe (but I am not if people are active there)?

Reply:
These discussions/definitions should be easy to have/solve, but it's not currently, especially because we're a big team. I don't know what you, or what other people think, but our asynchronous model to reach consensus in technical discussions hasn't been working very well for the last couple months. Github discussions don't get enough engagement either. You can try that @OmarBasem.

I suggest we bring this topic to the retrospective table and come up with something practical. I've worked in a team where we accumulated a certain backlog of small technical things, and every now and then we'd get together and make some decisions synchronously. The problem here is that many of these tiny discussions are not important enough to stop the entire team, so they easily get lost in a Github thread, like this one with only you and me.

Reply:
@ilmotta that's right. Btw, maybe you can talk note of these few things that you come across, then make a PR every now and then with some updates to the guidelines as suggestions, and request `status-mobile/devs` for review, that way you would bring everyone's attention, and also without having to bring several people into discussions on PRs repeteadly.

Reply:
I can try that @OmarBasem, took note ;) Thanks